plugin/helpers_config_test.go

```diff
@@ -0,0 +1,81 @@
+package plugin_test
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+	"github.com/mattermost/mattermost-server/plugin/plugintest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCheckRequiredServerConfiguration(t *testing.T) {
+	for name, test := range map[string]struct {
+		SetupAPI     func(*plugintest.API) *plugintest.API
+		Input        *model.Config
+		ShouldReturn bool
+		ShouldError  bool
+	}{
+		"no required config therefore it should be compatible": {
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("GetConfig").Return(nil)
+
+				return api
+			},
+			Input:        nil,
+			ShouldReturn: true,
+			ShouldError:  false,
+		},
+		"same configurations": {
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("GetConfig").Return(&model.Config{
+					ServiceSettings: model.ServiceSettings{
+						EnableCommands: model.NewBool(true),
+					},
+				})
+
+				return api
+			},
+			Input: &model.Config{
+				ServiceSettings: model.ServiceSettings{
+					EnableCommands: model.NewBool(true),
+				},
+			},
+			ShouldReturn: true,
+			ShouldError:  false,
+		},
+		"could not merge configurations": {
```

Comment:
This is actually, `different configurations`, right? Since there's no way to make `utils.Merge` fail?

Reply:
yeah, I think the naming convention should be changed with your suggestion :)